src/ProjectTemplates/Shared/Project.cs

```diff
@@ -30,14 +30,14 @@ public static string ArtifactsLogDir
     {
         get
         {
-            var testLogFolder = typeof(Project).Assembly.GetCustomAttribute<TestFrameworkFileLoggerAttribute>()?.BaseDirectory;
-            if (!string.IsNullOrEmpty(testLogFolder))
+            var helixWorkItemUploadRoot = Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT");
+            if (!string.IsNullOrEmpty(helixWorkItemUploadRoot))
             {
-                return testLogFolder;
+                return helixWorkItemUploadRoot;
             }
 
-            var helixWorkItemUploadRoot = Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT");
-            return string.IsNullOrEmpty(helixWorkItemUploadRoot) ? GetAssemblyMetadata("ArtifactsLogDir") : helixWorkItemUploadRoot;
+            var testLogFolder = typeof(Project).Assembly.GetCustomAttribute<TestFrameworkFileLoggerAttribute>()?.BaseDirectory;
```

Comment:
Nit: `TestFrameworkFileLoggerAttribute` should be added unconditionally and `BaseDirectory` should never be `null` based on a quick reading of Microsoft.AspNetCore.Testing.props. Might be time to at least remove the question mark here and `throw` if the attribute is missing for some strange reason.